src/main/java/TaskList.java

```diff
@@ -0,0 +1,186 @@
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> taskArrayList;
+
+    public TaskList() {
+        this.taskArrayList = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> taskArrayList) throws DukeException {
+        this.taskArrayList = taskArrayList;
+    }
+
+    /**
+     * This method takes an input string and formats it by including horizontal lines above
+     * and below the input string
+     *
+     * @param str input string to be sandwiched
+     * @return the original string sandwiched between two horizontal lines
+     */
+    public static String sandwich(String str) {
+        return "____________________________________________________________\n"
+                + str + "\n"
+                + "____________________________________________________________";
+    }
+
+    /**
+     * This method takes the user's input list and beautifies it for display.
+     *
+     * @param taskArrayList the user's input list to be beautified
+     * @return the beautified string to display
+     */
+    public String listBeautify(ArrayList<Task> taskArrayList) {
+        StringBuilder listBeautified = new StringBuilder();
+        for (int i = 0; i < taskArrayList.size(); i++) {
+            listBeautified.append(i + 1)
+                    .append(".")
+                    .append(taskArrayList.get(i).toString());
+            if (i < taskArrayList.size() - 1) { // new line except for last item
+                listBeautified.append("\n");
+            }
+        }
+        return listBeautified.toString();
+    }
+
+    /**
+     * Validates input string to ensure it follows the valid format YYYY-MM-DD, and is a valid date.
+     *
+     * @param input the string to be validated
+     * @return true if the string is a valid date
+     */
+    public boolean isValidDate(String input) {
```

Comment:
Might I suggest using an inbuilt Java method like DateTimeFormatter or LocalDateTime to replace isValidDate() to improve code readability?